frontend/src/main/components/BasicCourseSearch/CourseOverTimeSearchForm.js

```diff
@@ -70,17 +70,17 @@ const CourseOverTimeSearchForm = ({ fetchJSON }) => {
 
   const handleCourseNumberOnChange = (event) => {
     const rawCourse = event.target.value;
-    if (rawCourse.match(/\d+/g) != null) {
-      const number = rawCourse.match(/\d+/g)[0];
+    const firstDigitIndex = rawCourse.search(/\d/);
+    if (firstDigitIndex !== -1) {
+      const strippedCourse = rawCourse.slice(firstDigitIndex);
+      const numberMatch = strippedCourse.match(/\d+/g);
+      const suffixMatch = strippedCourse.match(/[a-zA-Z]+/g);
+      const number = numberMatch[0];
+      const suffix = suffixMatch ? suffixMatch[0] : "";
```

Comment:
An observation:  Suppose instead, you had written a utility function such as this:

```js
const getNumberSuffix = (rawCourse) = {
    if (firstDigitIndex !== -1) {
      const strippedCourse = rawCourse.slice(firstDigitIndex);
      const numberMatch = strippedCourse.match(/\d+/g);
      const suffixMatch = strippedCourse.match(/[a-zA-Z]+/g);
      const number = numberMatch[0];
      const suffix = suffixMatch ? suffixMatch[0] : "";
      return { number, suffix };
      } else {
        return { number: "", suffix: ""}
      }
}
```

Then you could have just used that, and I'm guessing the testing would be 10x easier.  You'd do most of the testing of all the different cases in the unit tests for the little pure javascript function, and the main test of the component could be much simpler.

You could still do that in another story for another 10 points though.